### PR TITLE
app/vmalert: initial remote-write support for alerts state persistence.

### PR DESCRIPTION
**Pull request by hagen1778**
If `remotewrite.url` flag is set, vmalert will send alerts state  via remote-write protocol to remote storage. The sending is asynchronous to avoid blocking calls in rules evaluation loop.

### CODE REVIEW

File: app/vmalert/remotewrite/remotewrite.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+	}
+	close(c.input)
+	close(c.doneCh)
+	c.wg.Wait()
+	return nil
+}
+
+func (c *Client) run(ctx context.Context) error {
+	c.input = make(chan prompbmarshal.TimeSeries, 100)
+	ticker := time.Tick(c.flushInterval)
```
Comment (tenmozes) [inline app/vmalert/remotewrite/remotewrite.go:106]: should it be time.Ticker?

File: app/vmalert/remotewrite/remotewrite.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	if c.doneCh == nil {
+		return fmt.Errorf("client is already closed")
+	}
+	close(c.input)
+	close(c.doneCh)
+	c.wg.Wait()
+	return nil
+}
+
+func (c *Client) run(ctx context.Context) error {
```
Comment (tenmozes) [inline app/vmalert/remotewrite/remotewrite.go:104]: it looks like error is always nil

File: app/vmalert/remotewrite/remotewrite.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	c.input = make(chan prompbmarshal.TimeSeries, 100)
```
Comment (tenmozes) [inline app/vmalert/remotewrite/remotewrite.go:105]: do we need to move 100 to config?

File: app/vmalert/remotewrite/remotewrite.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+// Push adds timeseries into queue for writing into remote storage.
+// Push returns and error if client is stopped or if queue is full.
+func (c *Client) Push(s prompbmarshal.TimeSeries) error {
+	select {
+	case <-c.doneCh:
+		return fmt.Errorf("client is closed")
+	case c.input <- s:
+		return nil
+	default:
+		return fmt.Errorf("failed to push timeseries - queue is full")
```
Comment (tenmozes) [inline app/vmalert/remotewrite/remotewrite.go:88]: can we add here information about queue capacity?

Comment (hagen1778) [reply app/vmalert/remotewrite/remotewrite.go]: Added

File: app/vmalert/remotewrite/remotewrite.go (hunk 3 above)
Comment (hagen1778) [reply app/vmalert/remotewrite/remotewrite.go]: Added

File: app/vmalert/remotewrite/remotewrite.go (hunk 1 above)
Comment (hagen1778) [reply app/vmalert/remotewrite/remotewrite.go:123]: I see no difference in this particular case.

**Pull request merged**